### ISSUE TITLE
Gradle should treat the _JAVA_OPTIONS environment variable as an implicit task input

### ISSUE THREAD

**Issue by staktrace** [gradle/gradle#26381]
### Current Behavior

Currently, gradle has a bunch of environment/system settings that it automatically considers to be task inputs. For example, the `test` task automatically includes the following (among many others):
```
Appending input value fingerprint for 'javaVersion' to build cache key: a40a171a9de3ed0d81e29d9af4cc27ef
Appending input value fingerprint for 'jvmArgs' to build cache key: 8222d82255460164427051d7537fa305
```
This is completely reasonable, as changing the JVM version or JVM arguments can impact how code is run, and affect the output of the tests.

However, gradle does not seem to treat the `_JAVA_OPTIONS` environment variable in the same way, but for consistency, it should - this environment variable is read by the JVM directly and can be used update the JVM settings. This can result in dramatically different behaviour during `test` (and maybe other) tasks, and should be treated as a task input.

### Expected Behavior

Changing the value of the `_JAVA_OPTIONS` environment variable should cause `test` (and possibly other) tasks to re-run rather than re-using previously cached results.

### Context (optional)

In our case, we changed our `_JAVA_OPTIONS` environment variable in CI to include a new java agent (via the `-javaagent` argument). Although this should have caused tests to fail, gradle re-used cached test output from a previous run and so the CI run passed. This later resulted in downstream breakage when the test was next re-run.

### Steps to Reproduce

I can try to create a minimal example, but our original example has quite a bit of proprietary code so I haven't yet. If it would be useful (i.e. if this ticket isn't immediately closed as WONTFIX) I'm happy to put in the effort to create a reproducible example.

### Gradle version

7.6.1

**Comment by big-guy** [comment]
`JAVA_TOOL_OPTIONS` is the public API equivalent: https://bugs.openjdk.org/browse/JDK-4971166

Another view of this is that Gradle should explicitly strip out this environment variable (and others like it) because it's a source for surprises.

**Comment by staktrace** [comment]
> `JAVA_TOOL_OPTIONS` is the public API equivalent: https://bugs.openjdk.org/browse/JDK-4971166

Fair enough, I'd be fine with that env var being treated as a task input instead of `_JAVA_OPTIONS`. I think fundamentally that doesn't alter the bug much.

> Another view of this is that Gradle should explicitly strip out this environment variable (and others like it) because it's a source for surprises.

I'm not sure this is desirable. The env var exists for a reason and serves a useful function. And besides, Gradle stripping it out would be a much bigger breaking change.

**Comment by jbartok** [comment]
Thank you for your interest in Gradle!

This issue needs a decision from the team responsible for that area. They have been informed. Response time may vary.

**Comment by asodja** [comment]
Note: You can workaround this, by adding that env variable as an input manually, for example:
```
tasks.named<Test>("test") {
    inputs.property("javaOptionsEnvVariable", project.providers.environmentVariable("_JAVA_OPTIONS"))
}
```

**Comment by staktrace** [comment]
Yes, it can be worked around, and we have done so. Still, I feel this should be fixed in Gradle itself.

**Comment by big-guy** [comment]
@staktrace Instead of relying on _JAVA_OPTIONS to pass the java agent to tests, would it be possible for you to set it for _just_ test tasks?

That would mean you would do something like:
```
test {
   inputs.files file("/path/to/javaagent.jar")
   jvmArgs "-javaagent:/path/to/javaagent.jar"
}
```
IIUC, the way you're using _JAVA_OPTIONS now will impact things other than tests: the Gradle daemon, Gradle workers (compilation, checkstyle, etc) and JavaExec.

**Comment by staktrace** [comment]
> IIUC, the way you're using _JAVA_OPTIONS now will impact things other than tests

This is the desired outcome. The java agent is an instrumentation agent and we want it to run in all cases, not just during test tasks. So to answer your question, no, just setting the agent on the test task wouldn't be appropriate for our use case.

**Comment by asodja** [comment]
This is a valid report.

The issue is in the backlog of the team, but this area of Gradle is currently not a focus one, so it might take a while before a fix is made.

The current workaround is to add the env variable as input to tasks manually.
